src/hotspot/share/opto/idealGraphPrinter.cpp

```diff
@@ -374,9 +374,26 @@ void IdealGraphPrinter::visit_node(Node *n, bool edges, VectorSet* temp_set) {
 #ifndef PRODUCT
     Compile::current()->_in_dump_cnt++;
     print_prop(NODE_NAME_PROPERTY, (const char *)node->Name());
+    print_prop("idx", node->_idx);
     const Type *t = node->bottom_type();
     print_prop("type", t->msg());
-    print_prop("idx", node->_idx);
+    if (t->category() != Type::Category::Control &&
+        t->category() != Type::Category::Memory) {
+      // Print detailed type information for nodes whose type is not trivial.
+      buffer[0] = 0;
+      stringStream bottom_type_stream(buffer, sizeof(buffer) - 1);
+      t->dump_on(&bottom_type_stream);
+      print_prop("bottom_type", buffer);
+      if (types != nullptr) {
```

Comment:
Could we just use `Compile::current()->types()` here instead of passing it in? We do not seem to use any other types array apart from `Compile::types()`. We are already accessing `Compile::current()` above on line 375.

Reply:
Thanks for reviewing, Christian! Good catch, turns out `IdealGraphPrinter` holds a pointer to the `Compile` instance (`Compile* IdealGraphPrinter::C`), so I just used that.